### DISCUSSION TITLE
wierd database discrepancy between binary and docker

### DISCUSSION THREAD

**Discussion by Barborica-Alexandru** [dani-garcia/vaultwarden#2472] [Q&A] [answered]
### Subject of the issue

Hello, I've been using vaultwarden by building the binary for years and have recently decided to switch to the docker container. In both cases I am using the postgresql backend. On my original database in the table devices there is a constraint devices_pkey that includes uuid and user_uuid. On the docker container version this contraint includes just uuid and not user_uuid which leads to a postgresql error when trying to log in because the query to add the device fails. The solution is to delete and recreate the constraint with just uuid as is on the freshly installed docker version. 

Im just curios how this happened? And could there be other discrepancies that could lead to issues if I just use my old database with the corrected constraint?
### Deployment environment
Binary and docker, migration done on latest versions on both platforms, same database.

**Comment by BlackDex** [comment, marked as answer]
I wonder what you mean with `latest` version.
If you mean there is a difference between the database schema regarding the `latest` **tagged docker** images and the `latest` **commit**, then yea there is a difference, since they do not match.

The `latest` **commit** equals the `testing` **tagged images**.
The `latest` **tagged** image equals the latest stable released version.

If you build from source your self or use the docker build images they should match 1:1 when you use the exact same commit hash (depending a little bit on which OS you build it because of libraries etc..).

**Reply by Barborica-Alexandru** [reply]
Yes this is probably the difference, I would just pull the repository and build up to now. So essentially I downgraded from testing to stable. Is there any issue with this aside from the devices table constraint difference?

Should I just create a new database and manually create users and import the vaults? Or should I switch to the testing docker images?

I checked my git version and yes I am up to date with origin/main so essentially testing. 

I switched to docker testing and it works just fine with my database schema so this was the issue. I think to switch over to stable I need to not update until stable catches up to testing. This is fine, I'm just glad I understand why it happened.

**Reply by BlackDex** [reply]
Good to see it is resolved.

The changes were not that big if I'm correct. And i tested switching back and forward a few times without issues.
